src/CollapseEmbeds/hider.tsx

```diff
@@ -15,11 +15,71 @@ export interface HiderProps {
     placeholders: string[];
     type: AccessoryType;
     children: React.ReactNode;
+    id?: string;
 }
 
-export const Hider = ({placeholders, type, children}: HiderProps): JSX.Element => {
-    const [shown, setShown] = React.useState(!Settings.current.hideByDefault);
-    Settings.useListener(({hideByDefault}) => setShown(!hideByDefault), []);
+export const Hider = ({placeholders, type, children, id}: HiderProps): JSX.Element => {
+    // Update lastSeen timestamp on mount and track component mounted state
+    const mounted = React.useRef(true);
+
+    React.useEffect(() => {
+        mounted.current = true;
+        
+        if (id) {
+            const state = Settings.current.collapsedStates[id];
+            if (state) {
+                const newStates = {
+                    ...Settings.current.collapsedStates,
+                    [id]: { ...state, lastSeen: Date.now() }
+                };
+                Settings.update({
+                    ...Settings.current,
+                    collapsedStates: newStates
+                });
+            }
+        }
+
+        return () => {
+            mounted.current = false;
+        };
+    }, [id]);
+
+    const [shown, setShown] = React.useState(() => {
+        if (!id) return !Settings.current.hideByDefault;
```

Comment:
Under what circumstances is the id prop = url expected to be empty? I think what you meant do do is check whether there is no entry in collapsed states yet.

Reply:
This is not resolved yet. The condition is never true, so the hide by default setting is always ignored and not functional.